lib/AFS/CellCC/Dump.pm

```diff
@@ -309,32 +351,78 @@ process_dumps($$$@) {
                              timeout => 3600,
                              description => "Waiting for dump to be scheduled");
 
+    # Group the jobs with the same volume_name and timestamp
+    $start_time = time();
     for my $job (@jobs) {
-        my $pid = $pm->start();
-        if ($pid) {
-            # In parent
-            DEBUG "Spawned child pid $pid to handle dump for job ".$job->{jobid};
-            next;
+        eval {
+            my $volname = $job->{volname};
+            my $lastupdate = _calc_incremental($job, $start_state);
```

Comment:
I really really do not want to run `_calc_incremental` directly inside `process_dumps` if we don't have to... that will prevent all other dumps from processing until it finishes running and can let someone else come and "steal" the jobs out from under us; that's why we spawn all of the real work in child procs. But of course we can't just spawn this block in a child proc, because we need to access the global %vol_table. Instead what we can do that I think is still acceptable is create a child proc per volume name, and have that proc run `_calc_incremental` for each job for that volume name. That can just populate the vol_lastupdate field in the db and change the state, and `_process_dumps` can find all of the jobs in that state and populate %vol_table from that data.

Do you think that can work? We'll still serialize `_calc_incremental` calls for different jobs for the same volumes, but jobs for different volumes won't hold each other up.

Reply:
>Instead what we can do that I think is still acceptable is create a child proc per volume name

Or no, that doesn't make much sense. If we're going to do this, just create a child proc per job like normal, and have that proc fill in `vol_lastupdate` in the db. Wait until all jobs are done doing that, and just proceed like normal; the parent process can then fill in the `%vol_table` table from data that's in the db.